addons/website_slides_survey/models/slide_channel.py

```diff
@@ -1,13 +1,56 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
 from odoo.osv import expression
+from odoo.tools import SQL
 
+class ChannelUsersRelation(models.Model):
+    _inherit = 'slide.channel.partner'
+
+    certification_passed = fields.Boolean('Certified', compute='_compute_certification_passed', search='_search_certification_passed')
+    nbr_certification = fields.Integer(related='channel_id.nbr_certification')
+
+    @api.depends('channel_id.slide_partner_ids.survey_scoring_success')
+    def _compute_certification_passed(self):
+        base_domain = expression.OR([
+            [('channel_id', '=', slide_channel.channel_id.id),
+             ('partner_id', '=', slide_channel.partner_id.id)]
+            for slide_channel in self
+        ])
+        certified = self.env['slide.slide.partner'].sudo()._read_group(
+            domain=expression.AND([base_domain, [('survey_scoring_success', '=', True)]]),
+            groupby=['channel_id'],
+            aggregates=['partner_id:recordset']
+        )
+        mapped_data = {channel_id: partner_ids for channel_id, partner_ids in certified}
+        for member in self:
+            member.certification_passed = member.partner_id in mapped_data.get(member.channel_id, [])
```

Comment:
Would this work?
```suggestion
        certified_members = self.search([['id', 'in', self.ids], self._search_certification_passed('=', True)])
        certified_members.certification_passed = True        
        (self - certified_members).certification_passed = False
```

Reply:
Python test passes, functional test also ok. Just needed to use expression to combine the two lists of tuple as a domain (got this error with the provided syntax: IndexError: tuple index out of range).